features/layout/sidebar-navigation/sidebar-navigation.tsx

```diff
@@ -21,6 +21,30 @@ export function SidebarNavigation() {
   const { isSidebarCollapsed, toggleSidebar } = useContext(NavigationContext);
   const [isMobileMenuOpen, setMobileMenuOpen] = useState(false);
 
+  //State for if it's mobile view or not and accounts for the server side rendering delay
+  const [isMobileView, setIsMobileView] = useState(
+    typeof window !== "undefined" ? window.innerWidth <= 1023 : false,
+  );
+
+  useEffect(() => {
+    const handleWindowResize = () => {
+      setIsMobileView(window.innerWidth <= 1023);
+    };
```

Comment:
An alternative to checking the window width would be to use [matchMedia](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia). There's actually [a ready-made hook in on GitHub](https://github.com/juliencrn/usehooks-ts/blob/master/packages/usehooks-ts/src/useMediaQuery/useMediaQuery.ts) that could act as inspiration 😄

Reply:
And btw I usually try to get around JS media queries where possible and use CSS instead